### ISSUE TITLE
EFR breaks spawn mobs

### ISSUE THREAD

**Issue by VadimFlex** [Roadhog360/Et-Futurum-Requiem#424]
### Initial Questions

- [X] I have searched this issue tracker and there is nothing similar already, and this is not on the [list of known issues](https://github.com/Roadhog360/Et-Futurum-Requiem/issues/146). Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I have verified that this issue occurs in a SUPPORTED environment, meaning I can reproduce this WITHOUT OptiFine, and without Bukkit+Forge server software like KCauldron, Thermos, Crucible, Mohist etc

### Mod Version

Et_Futurum_Requiem-2.6.0-SNAPSHOT-7e8d829

### Describe the Issue

EFR breaks the spawn of hostile mobs on the surface. There is no such problem in caves. With Optifine installed, it's even worse. This applies to vanilla mobs and mobs from DivineRPG. There was no such problem in older versions.

**Comment by makamys** [comment]
The modern spawning logic that makes hostile mobs only spawn at light level 0 was backported recently, sounds like it's working as intended. You can turn it off in the config, the option is called `enableLightLevel0`.

**Comment by GregoriusT** [comment]
Question about that, this is overworld only right? I really hope it is Overworld only because in all other dimensions this can result in huge Issues.

**Comment by Roadhog360** [comment]
DivineRPG has been known to break my spawning behavior. OptiFine probably also breaks the behavior too.

**Comment by Roadhog360** [comment]
> The modern spawning logic that makes hostile mobs only spawn at light level 0 was backported recently, sounds like it's working as intended. You can turn it off in the config, the option is called `enableLightLevel0`.

No, the monster should be spawning on the surface too. I haven't tested this specific change yet, but I'll see about reproduction steps.

P.S. To VadimFlex I know you said that you are able to reproduce the issue without OptiFine but if you are using it outside of bug reporting, I suggest you don't because it is broken. If you just need the optimizations there are other great optimization mods that can do this for you.

**Comment by GregoriusT** [comment]
Oh but i hope you did only do it to vanilla mobs and EFR mobs, otherwise Lycanites and other mods will not be able to spawn certain Lava specific Mobs for example.

**Comment by VadimFlex** [comment]
> Question about that, this is overworld only right? I really hope it is Overworld only because in all other dimensions (and I really mean ALL of them!) this can result in huge Issues.

Unfortunately, to one degree or another, this also applies to the worlds of DivineRPG

> Oh but i hope you did only do it to vanilla mobs and EFR mobs, otherwise Lycanites and other mods will not be able to spawn certain Lava specific Mobs for example.

This also applies to mobs from DivineRPG

**Comment by VadimFlex** [comment]
> The modern spawning logic that makes hostile mobs only spawn at light level 0 was backported recently, sounds like it's working as intended.

I don't quite understand, but is there something like this in the new versions of minecraft? In version 1.20.1, hostile mobs spawned on the surface



> If you just need the optimizations there are other great optimization mods that can do this for you, usually to a greater effect.

What can you recommend? In fact, I use Optifine only for the "Improved Snow" feature, which adds visual snow to flowers, grasses, fences

**Comment by Roadhog360** [comment]
> Oh but i hope you did only do it to vanilla mobs and EFR mobs, otherwise Lycanites and other mods will not be able to spawn certain Lava specific Mobs for example.

Lava mobs won't be using the default mob spawner, so this should be OK.

> What can you recommend? In fact, I use Optifine only for the "Improved Snow" feature, which adds visual snow to flowers, grasses, fences

I'm unaware of OptiFine introducing such a feature. Can you show me? As for what I recommend. [here](https://gist.github.com/makamys/7cb74cd71d93a4332d2891db2624e17c#optimizationfix-mods). I don't think we have improved snow yet though but using mods from here should be far more stable and effective than the broken OptiFine

**Comment by Roadhog360** [comment]
Might remove the light level 0 config or have it disabled for the time being, since it'll probably unintentionally block the spawning of mobs that are not meant to spawn in darkness. Greg brings up a good point. We also have the issue of the mobs on the surface not spawning, which I am unsure as to the cause of.
I don't want to just target vanilla mobs because then that will introduce inconsistent spawning behavior between different darkness-based baddies that will be difficult for the player to reasonably be able to distinguish.


This behavior will likely need some fine tuning to only target mobs that are meant to actually spawn in the darkness. I might rather mixin the mob spawning logic instead to change the light value it checks for. Greg brings up a good point and while at first I thought the logic currently used would work, it probably will not in the long run. Closing as resolved, I am removing this feature and replacing it with a mixin directly into the darkness check in the mob spawning code at a future date.

**Comment by makamys** [comment]
> I don't quite understand, but is there something like this in the new versions of minecraft? In version 1.20.1, hostile mobs spawned on the surface

I seem to have misinterpreted your report, I didn't realize mobs weren't spawning on the surface *at all*, I thought they were just spawning less near light sources (which is what the feature was supposed to do).

**Comment by GregoriusT** [comment]
If they dont spawn on the surface at all, then you MIGHT have grabbed the mixed light level which is pretty much always above zero with the skylight, same for Nether and End where it is locked at i think 7 or something, which would block skeletons from nether fortresses for example.

**Comment by VadimFlex** [comment]
> I'm unaware of OptiFine introducing such a feature. Can you show me?

![2024-02-14_22 58 50](https://github.com/Roadhog360/Et-Futurum-Requiem/assets/132595752/2510c470-c9fb-42e5-b34d-56bf369bfddf)
![2024-02-14_22 59 46](https://github.com/Roadhog360/Et-Futurum-Requiem/assets/132595752/9a30e877-4fc6-4305-a132-6c563a2bb7b3)


> As for what I recommend. [here](https://gist.github.com/makamys/7cb74cd71d93a4332d2891db2624e17c#optimizationfix-mods).

Thanks)

**Comment by Roadhog360** [comment]
Oh, that does look quite nice, I'll see if it can be implemented in Angelica in the future. While Angelica is on the list it is still in mid-Alpha so it isn't quite stable yet, but I figured I'd mention the option in case it works in your setup; it's supposed to be a better OptiFine replacement.

**Comment by VadimFlex** [comment]
> Oh, that does look quite nice, I'll see if it can be implemented in Angelica in the future.

I hope this is possible to implement) 
This Optifine function works well, except that it doesn't work for some plants from BOP, as well as flowers from Botany
